File: frontend/src/lib/entities/memory.ts
export interface Memory {
	id: string;
	content: string;
	enabled: boolean;
	metadata?: Record<string, any> | null;
	created_at?: string;
	updated_at?: string;
}

export interface MemoryListResponse {
	memories: Memory[];
	total: number;
	limit: number;
	offset: number;
}

export interface MemoryCreateRequest {
	content: string;
	path: string;
	metadata?: Record<string, any> | null;
}

export interface MemoryUpdateRequest {
	content: string;
	metadata?: Record<string, any> | null;
	enabled?: boolean;
}

File: backend/src/schemas/entities/memory.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class Memory(BaseModel):
    id: str
    content: str
    enabled: bool = True
    metadata: Optional[dict] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class MemoryCreate(BaseModel):
    content: str = Field(..., min_length=1)
    path: str = Field(..., min_length=1)
    metadata: Optional[dict] = None


class MemoryUpdate(BaseModel):
    content: str = Field(..., min_length=1)
    metadata: Optional[dict] = None
    enabled: Optional[bool] = None


class MemoryListResponse(BaseModel):
    memories: list[Memory]
    total: int
    limit: int
    offset: int
